dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGVisualizeLauncher.java

```diff
@@ -0,0 +1,186 @@
+package org.checkerframework.dataflow.cfg;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.dataflow.analysis.AbstractValue;
+import org.checkerframework.dataflow.analysis.Analysis;
+import org.checkerframework.dataflow.analysis.Store;
+import org.checkerframework.dataflow.analysis.TransferFunction;
+
+/** The CFG visualize launcher. */
+public class CFGVisualizeLauncher {
+
+    /** Main method. */
+    public static void main(String[] args) {
+        if (args.length < 2) {
+            printUsage();
+            System.exit(1);
+        }
+        String input = args[0];
+        String output = args[1];
+        File file = new File(input);
+        if (!file.canRead()) {
+            JavaSource2CFG.printError("Cannot read input file: " + file.getAbsolutePath());
+            printUsage();
+            System.exit(1);
+        }
+
+        String method = "test";
+        String clas = "Test";
+        boolean pdf = false;
+        boolean error = false;
+        boolean verbose = false;
+
+        for (int i = 2; i < args.length; i++) {
+            switch (args[i]) {
+                case "-pdf":
+                    pdf = true;
+                    break;
+                case "-method":
+                    if (i >= args.length - 1) {
+                        // TODO: extract this static util method out to Util class if we accumulate
+                        //  enough number of this kind of util methods
+                        JavaSource2CFG.printError("Did not find <name> after -method.");
+                        continue;
+                    }
+                    i++;
+                    method = args[i];
+                    break;
+                case "-class":
+                    if (i >= args.length - 1) {
+                        JavaSource2CFG.printError("Did not find <name> after -class.");
+                        continue;
+                    }
+                    i++;
+                    clas = args[i];
+                    break;
+                case "-verbose":
+                    verbose = true;
+                    break;
+                default:
+                    JavaSource2CFG.printError("Unknown command line argument: " + args[i]);
+                    error = true;
+                    break;
+            }
+        }
+
+        if (error) {
+            System.exit(1);
+        }
+
+        generateDOTofCFGWithoutAnalysis(input, output, method, clas, pdf, verbose);
+    }
+
+    /** Generate DOT representation of CFG graph without analysis. */
+    public static void generateDOTofCFGWithoutAnalysis(
```

Comment:
Instead of making all these `public static` could you make them protected instance methods to allow subclasses?